src/charm.py

```diff
@@ -158,6 +159,12 @@ def _configure(self, _):
         ):
             return
 
+        # On upgrade, make sure the remote write endpoint is changed. Note that
+        # this is only needed for upgrade from previous versions that
+        # use bind_address rather than the fqdn. If at some point we break the
+        # upgrade path, we should look into removing this.
+        self.remote_write_provider.update_endpoint()
```

Comment:
Yeah but then shouldn’t we rather do it in an upgrade_charm hook?

Reply:
This is the upgrade charm hook

Reply:
This is almost every hook ;-)

Wouldn’t it make sense to add an intermediate `_on_charm_upgrade(self, event)` method that does this invocation and then delegates to `self._configure(event)`?

Reply:
Yes but I didn't not want to so aggressively violate the design decisions of the original author. I will bring it up in the next meeting.

Reply:
Upon more inspection, I realized that this is needed in `upgrade_charm`, `pebble_ready` (for pod churn), and `ingress_changed`. So I just fixed the comment and kept it in `_configure`.

Reply:
I am wondering -- "if it makes sense for the Ingress object listen to pod churn events (`RelationJoined`, `RelationDeparted` etc) and raise `IngressChanged` events".